qiskit_ibm_runtime/accounts/account.py

```diff
@@ -13,16 +13,32 @@
 """Account related classes and functions."""
 
 
-from typing import Optional
+from typing import Optional, Dict
 from urllib.parse import urlparse
 
-from typing_extensions import Literal
 from requests.auth import AuthBase
-
+from typing_extensions import Literal, TypedDict
+from ..utils.hgp import from_instance_format
 from ..api.auth import LegacyAuth, CloudAuth
 
 AccountType = Optional[Literal["cloud", "legacy"]]
 
+
+class ProxyConfigurationType(TypedDict, total=False):
+    """Dictionary type for custom proxy configuration.
+
+    All items in the dictionary are optional. When ``urls`` are provided, they must contain a dictionary
+    mapping protocol or protocol and host to the URL of the proxy. Refer to
+    https://docs.python-requests.org/en/latest/api/#requests.Session.proxies for details and examples.
+
+    NTLM user authentication can be enabled by setting ``username_ntlm`` and ``password_ntlm``.
+    """
+
+    urls: Dict[str, str]
+    username_ntlm: str
+    password_ntlm: str
```

Comment:
The proxy configuration dictionary is passed to multiple public and private APIs. I did not want to duplicate more detailed documentation in the docstrings across all places.

I therefore decided to add a type hint for the existing dictionary (see [PEP-0589](https://www.python.org/dev/peps/pep-0589/)) that will also add type safety internally.

Reply:
Making it a `TypedDict` is an interesting idea. However, you'd still still need to document it in the user facing function (`IBMRuntimeService` constructor) _or_ make the hyperlink work so a user can click on the link and see the docstring for this class.

An alternative idea is to make it a normal class. You document the usage in `IBMRuntimeService` constructor and then convert it to your new class. The benefit of that is you can then add methods like `validate()` and `to_request_params()`, which encapsulate all the work related to proxy specifications. Unfortunately it doesn't seem custom methods are supported for `TypedDict` subclasses. 

In addition, I'd suggest making it a separate module instead of lumping into `account.py` since its content is used by both `Account` and `ClientParams`.

Reply:
Thanks for your thoughtful and crisp suggestions. Updated code accordingly and think/hope it got much cleaner, @jyu00. Please take a look.